interfaces/builtin/microstack_support.go

```diff
@@ -0,0 +1,245 @@
+# as libvirt generates AppArmor profiles for the utility processes it spawns.
+/sys/kernel/security/apparmor/{,**} r,
+/sys/kernel/security/apparmor/.remove w,
+/sys/kernel/security/apparmor/.replace w,
+
+# Used by libvirt to work with IOMMU.
+/sys/kernel/iommu_groups/ r,
+/sys/kernel/iommu_groups/** r,
+/sys/bus/pci/devices/**/iommu_group/** r,
+
+# Used by libvirt's QEMU driver state initialization code path.
+# The path used is hard-coded in libvirt to <huge-page-mnt-dir>/libvirt/qemu.
+/dev/hugepages/libvirt/ rw,
+/dev/hugepages/libvirt/** mrwklix,
+
+# Used by QEMU to get the maximum number of memory regions allowed in the vhost kernel module.
+/sys/module/vhost/parameters/max_mem_regions r,
+
+# Used by libvirt (cgroup-related):
+/sys/fs/cgroup/unified/cgroup.controllers r,
+/sys/fs/cgroup/cpuset/cpuset.cpus r,
+
+# Non-systemd layout: https://libvirt.org/cgroups.html#currentLayoutGeneric
+/sys/fs/cgroup/*/ r,
+/sys/fs/cgroup/*/machine/ rw,
+/sys/fs/cgroup/*/machine/** rw,
+
+# systemd-layout: https://libvirt.org/cgroups.html#systemdLayout
+/sys/fs/cgroup/*/machine.slice/machine-qemu*/{,**} rw,
+
+@{PROC}/[0-9]*/cgroup r,
+@{PROC}/cgroups r,
+
+# Used by libvirt.
+@{PROC}/cmdline r,
+@{PROC}/filesystems r,
+@{PROC}/mtrr w,
+@{PROC}/@{pids}/environ r,
+@{PROC}/@{pids}/sched r,
+
+@{PROC}/*/status r,
+
+# Libvirt needs access to the PCI config space in order to be able to reset devices.
+/sys/devices/pci*/**/config rw,
+
+# Used by libvirt.
+/dev/ptmx rw,
+# spice
+owner /{dev,run}/shm/spice.* rw,
+
+# Used by libvirt to create lock files for /dev/pts/<num> devices
+# when handling virsh console access requests.
+/run/lock/ r,
+/run/lock/LCK.._pts_* rwk,
+
+# Used by LVM tools.
+/run/lock/lvm/ rw,
+/run/lock/lvm/** rwk,
+# Files like /run/lvm/pvs_online, /run/lvm/vgs_online, /run/lvm/hints
+/run/lvm/ rw,
+/run/lvm/** rwlk,
+/run/dmeventd-client rwlk,
+/run/dmeventd-server rwlk,
+
+# Used by targetcli tools to work with LIO.
+/sys/kernel/config/target/ rw,
+/sys/kernel/config/target/** rw,
+
+# Used by targetcli.
+/run/targetcli.lock rwlk,
+/var/run/targetcli.lock rwlk,
+
+# Paths accessed by iscsid during its operation.
+/run/lock/iscsi/ rw,
+/run/lock/iscsi/** rwlk,
+/sys/devices/virtual/iscsi_transport/tcp/** r,
+/sys/devices/virtual/iscsi_transport/iser/** r,
+/sys/class/iscsi_session/** rw,
+/sys/class/iscsi_host/** r,
+/sys/devices/platform/host*/scsi_host/host*/** rw,
+/sys/devices/platform/host*/session*/connection*/iscsi_connection/connection*/** rw,
+/sys/devices/platform/host*/session*/iscsi_session/session*/** rw,
+/sys/devices/platform/host*/session*/target*/** rw,
+/sys/devices/platform/host*/iscsi_host/host*/** rw,
+
+# While the block-devices interface allows rw access, Libvirt also needs to be able to lock those.
+/dev/sd{,[a-h]}[a-z] rwk,
+/dev/sdi[a-v] rwk,
+# os-brick needs access to those when detaching a scsi device from an instance.
+/sys/block/sd{,[a-h]}[a-z]/device/delete rw,
+/sys/block/sdi[a-v]/device/delete rw,
```

Comment:
Don't you get the same effect if you modify the line above to use the `[a-i]` pattern? I see that in this way you are excluding `sdiz` -- is it some kind of special device?

Reply:
OK, seems legit :-)